src/03-a-two-state-system-in-a-quantised-field.md

```diff
@@ -0,0 +1,602 @@
+```
+
+```python
+a  = tensor(destroy(max_bosons+1), qeye(2))     # tensorised boson destruction operator
+sx = tensor(qeye(max_bosons+1), sigmax())       # tensorised sigma_x operator
+sz = tensor(qeye(max_bosons+1),sigmaz())        # tensorised sigma_z operator
+
+two_state     =  A*sz                      # two state system energy
+bosons       =  omega*(a.dag()*a+0.5)      # bosons field energy
+interaction   = V*(a.dag() + a) * sx       # interaction energy
+
+H = two_state + bosons + interaction
+```
+
+Because we've got 10 possible states, the probability plots that we normally create are going to get a bit crowded. To make plots easier to understand, we'll create a function to label the simulation data according to $\pm$ notation that we've been using up to this point to describe the two state system.
+
+```python
+def states_to_df(states,times):
+    data = {}
+    for i in range(0,states[0].shape[0]):
+        which_mode = divmod(i,2)
+        if which_mode[1] == 0:
+            two_state = "+"
+        elif which_mode[1] == 1:
+            two_state = "-"
+        data[str(which_mode[0])+" , "+two_state] = np.zeros(len(times),dtype="complex128")
+    
+    for i, state in enumerate(states):
+        for j, psi in enumerate(state):
+            which_mode = divmod(j,2)
+            if which_mode[1] == 0:
+                two_state = "+"
+            elif which_mode[1] == 1:
+                two_state = "-"
+            data[str(which_mode[0])+" , "+two_state][i] = psi[0][0]
+
+    return pd.DataFrame(data=data, index=times)
+```
+
+The quintessential set-up for spontaneous emission is to have the field empty, i.e. no bosons $n=0$, and the two state system in it's "excited" (aka higher energy "+") state.
+
+Let's see what happens.
+
+```python
+psi0 = tensor(basis(max_bosons+1, 0), basis(2, 0))  # No bosons and two-state system is in the higher energy + state
+
+times = np.linspace(0.0, 70.0, 1000)      # simulation time
+
+result = sesolve(H, psi0, times)
+df_coupled =  states_to_df(result.states, times)
+```
+
+```python
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15,6))
+df_coupled.plot(title="Real part of amplitudes Re($\psi$)     (Fig 1)", ax=axes[0]);
+(df_coupled.abs()**2).plot(title="Probabilities $|\psi|^2$     (Fig 2))", ax=axes[1]);
+```
+
+It's taken us a while but we eventually arrived, as promised. We see in Fig 2 that even though we start with the field "empty" of bosons (blue line), after a while, the field is likely to have a single boson at the expense of the energy in the two state system which transitions to the lower state (red line) - the atom "spontaneously" "emits" a boson.
+
+
+You might also notice some other features:
+1. There is non-negligible chance to find the system in what appears to be a non-energy conserving state with 2 bosons and the two state system in the + state (purple line)
+2. The spontaneous emission doesn't appear to be irreversible (as we are normally taught)
```

Comment:
Is this system reversible because it is actually the hamiltonian of a **cavity?**
Maybe we should specify this.
Actually things gets complicated here because a cavity contains a discrete, but infinite, number of modes. 

I'm develping more questions just while writing: what is the difference between the Hamiltonian of a cavity with a single mode, and a free space with a single mode?

Reply:
I think everything is reversible if you wait long enough. We see reversibility because the number of possible "microstates" for a given system energy  isn't very large. When we have mode modes it's a bit like having more particles of gas in a box - there are many more ways for the particles to be spread out than there are for the particles to be in a corner....so when you look in the box you most often find the system all spread out....but if you wait long enough you'll find them all back in the corner again.

Reply:
Re Hamiltonian - I don't think there is a difference. I think what's technically done in QED is to start with a box and gradually make the walls further and further apart until you approach a continuum

Reply:
well my thought was much more naive here: if a two state system emits a photon, than the photon will never find again the two-state system unless you have reflective walls. 
I said cavity because that means that the photon is delocalized all over the cavity I think, but if you have free space I don' t really know what it means and if the Hamiltonian looks the same

Reply:
Ah, yes I see what you mean. Let's discuss more in a bit